.github/workflows/pdm-lock-automation.yaml

```diff
@@ -0,0 +1,85 @@
+name: Automate pdm.lock
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review, review_requested]
+    branches: [main]
```

Comment:
Does this branches indicate the branch against which the PR was raised @kirtimanmishrazipstack ?

Reply:
yes @gaya3-zipstack this workflow will run when PR is made against `main`.
Sharing a link for your reference https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request